### ISSUE TITLE
nextcloud client: seg fault

### ISSUE THREAD

**Issue by r5r3** [r5r3/localflock#2]
```
Program received signal SIGSEGV, Segmentation fault.
0x0000155554ff1ecc in spdlog::logger::should_log(spdlog::level::level_enum) const () from /software/opt/bionic/x86_64/localflock/0.1/lib/liblocalflock.so
(gdb) quit
```

Seems like an problem related to the logger...

**Comment by r5r3** [comment]
```
(gdb) backtrace
#0  0x0000155554ff1ed2 in std::__atomic_base<int>::load(std::memory_order) const (__m=std::memory_order_relaxed, this=0x50)
    at /software/opt/bionic/x86_64/gcc/9.1.0/include/c++/9.1.0/bits/atomic_base.h:419
#1  0x0000155554ff1ed2 in spdlog::logger::should_log(spdlog::level::level_enum) const (this=0x0, msg_level=spdlog::level::debug) at /usr/include/spdlog/details/logger_impl.h:305
#2  0x0000155554ffb58b in spdlog::logger::log<int, int>(spdlog::level::level_enum, char const*, int const&, int const&) (this=0x0, lvl=spdlog::level::debug, fmt=0x1555550c619f "fcntl({0}, {1}, ...)", args#0=@0x7fffffffc9ec: 3, args#1=@0x7fffffffc9e8: 2) at /usr/include/spdlog/details/logger_impl.h:63
#3  0x0000155554ff715a in spdlog::logger::debug<int, int>(char const*, int const&, int const&) (this=0x0, fmt=0x1555550c619f "fcntl({0}, {1}, ...)", arg1=@0x7fffffffc9ec: 3, args#0=@0x7fffffffc9e8: 2) at /usr/include/spdlog/details/logger_impl.h:139
#4  0x0000155554febc6d in fcntl(int, int, ...) (fd=3, operation=2) at /home/r/Robert.Redl/Programm/localflock/src/localflock.cpp:129
#5  0x00001555484bcf2f in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#6  0x00001555484bd8d8 in QHashData::detach_helper(void (*)(QHashData::Node*, void*), void (*)(QHashData::Node*), int, int) () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#7  0x00001555485c1d61 in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#8  0x000015553cee7ef7 in  () at /usr/lib/x86_64-linux-gnu/libQt5Sensors.so.5
#9  0x000015555533c733 in call_init (env=0x7fffffffcca8, argv=0x7fffffffcc98, argc=1, l=<optimized out>) at dl-init.c:72
#10 0x000015555533c733 in _dl_init (main_map=0x155555555170, argc=1, argv=0x7fffffffcc98, env=0x7fffffffcca8) at dl-init.c:119
#11 0x000015555532d0ca in _dl_start_user () at /lib64/ld-linux-x86-64.so.2
#12 0x0000000000000001 in  ()
#13 0x00007fffffffd23c in  ()
#14 0x0000000000000000 in  ()

```

**Comment by r5r3** [comment]
The init function was not guaranteed to be called. Apparently an issue with multi-threaded applications.  This is fixed in commit c2b2e8a6256077cf968969e57eace3122eb300e1.
